source/jsonnet/england-wales/ccs/blocks/who-lives-here/anyone_else_list_collector.jsonnet

```diff
@@ -182,37 +181,98 @@ local editQuestion(questionTitle) = {
   add_block: {
     id: 'add-person',
     type: 'ListAddQuestion',
-    question: {
-      id: 'add-question',
-      type: 'General',
-      title: {
-        text: 'Who do you need to add?',
-        placeholders: [
-          placeholders.address,
-        ],
-      },
-      instruction: 'Enter a full stop (.) if the respondent does not know a person’s “First name” or “Last name”',
-      answers: [
-        {
-          id: 'first-name',
-          label: 'First name',
-          mandatory: true,
-          type: 'TextField',
+    question_variants: [
+      {
+        question: {
+          id: 'add-question',
+          type: 'General',
+          title: 'What is your full name?',
+          answers: [
+            {
+              id: 'first-name',
+              label: 'First name',
+              mandatory: true,
+              type: 'TextField',
+            },
+            {
+              id: 'middle-names',
+              label: 'Middle names',
+              mandatory: false,
+              type: 'TextField',
+            },
+            {
+              id: 'last-name',
+              label: 'Last name',
+              mandatory: true,
+              type: 'TextField',
+            },
+          ],
         },
-        {
-          id: 'middle-names',
-          label: 'Middle names',
-          mandatory: false,
-          type: 'TextField',
+        when: [rules.listIsEmpty('household'), rules.doesntLiveAtHouseholdAddress],
+      },
+      {
+        question: {
+          id: 'add-question',
+          type: 'General',
+          title: 'Who do you need to add?',
+          answers: [
+            {
+              id: 'first-name',
+              label: 'First name',
+              mandatory: true,
+              type: 'TextField',
+            },
+            {
+              id: 'middle-names',
+              label: 'Middle names',
+              mandatory: false,
+              type: 'TextField',
+            },
+            {
+              id: 'last-name',
+              label: 'Last name',
+              mandatory: true,
+              type: 'TextField',
+            },
+          ],
         },
-        {
-          id: 'last-name',
-          label: 'Last name',
-          mandatory: true,
-          type: 'TextField',
+        when: [rules.listIsNotEmpty('household'), rules.doesntLiveAtHouseholdAddress],
+      },
+      {
+        question: {
+          id: 'add-question',
+          type: 'General',
+          title: {
+            text: 'Who do you need to add?',
+            placeholders: [
+              placeholders.address,
+            ],
+          },
+          instruction: 'Enter a full stop (.) if the respondent does not know a person’s “First name” or “Last name”',
```

Comment:
On overflow, this is also shown for the first list collector. Can you verify what should be the correct behaviour?

Reply:
I followed the card:
"New driver question to route to a **new variant add answer screen** on anyone-else-list collector (New variant- 'What is your full name?'). **New variant should not have orange instruction panel**.

Variant when list empty- 'What is your full name?'
Variant when list not empty- 'Who do you need to add?'"

Did you check [overflow](https://overflow.io/s/FQ1RSI?node=3bf45946) attached to this card?

Reply:
The new variant shouldn't have it and that is correct but for the variant `Who do you need to add?` it should be displayed according to Overflow. On the quick launch, for the question `Who do you need to add?` the panel isn't displayed.

Reply:
Ok, I thought it is a new variant with two options, one of them is "when list not empty- 'Who do you need to add?'"?? I'll check with Donna.

Reply:
Done, thanks!👍